domain/secretbackend/state/state.go

```diff
@@ -361,8 +426,17 @@ func (s *State) ListSecretBackendsForModel(ctx context.Context, modelUUID coremo
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	// TODO: implement for inUse. JUJU-5707
-	stmt, err := s.Prepare(`
+
+	inUseCondition := ""
+	var inputArgs []any
+	if !includeEmpty {
+		inputArgs = append(inputArgs, SecretBackendReference{ModelID: modelUUID})
+		inUseCondition = fmt.Sprintf(`
+    LEFT JOIN secret_backend_reference sbr ON b.uuid = sbr.secret_backend_uuid
+WHERE sbr.model_uuid = $SecretBackendReference.model_uuid OR b.name = '%s' OR b.name = '%s'`[1:], kubernetes.BackendName, juju.BackendName)
```

Comment:
To avoid literal query params, can we change this to `b.name IN ($backendNames[:])`

Reply:
As discussed, we are ok for it because they are constants defined internally.